send_message_to_msteams.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import random
+from abc import ABC, abstractmethod
+
+import pymsteams
+import requests
+from requests.exceptions import ConnectionError, HTTPError
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+# Constants
+QUOTE_URL = "https://api.quotable.io/random"
+IMAGE_URL = "https://picsum.photos"
+WIDTH = random.randint(300, 800)
+HEIGHT = random.randint(300, 800)
+MESSAGE_COLOR = "#F8C471"
+TEAMS_WEBHOOK_URL = "your_webhook"  # TODO: store credentials in a secure way
+
+class ContentGenerator(ABC):
+    @abstractmethod
+    def get_content(self):
+        """ Method create connection to API and select random image/quote/etc."""
+        pass
+
+
+class QuoteGenerator(ContentGenerator):
+    def get_content(self):
+        response = requests.get(QUOTE_URL)
+
+        if response.status_code != 200:
+            logging.error("Failed to retrieve a quote")
+            return None
+        else:
+            data = response.json()
+            quote_text = data["content"]
+            quote_author = data["author"]
+            logging.info("A quote successfully generated.")
+
+            return quote_text, quote_author
+
+
+class ImageGenerator(ContentGenerator):
+
+    def get_content(self):
+        image_url = f"{IMAGE_URL}/{WIDTH}/{HEIGHT}"
+        return image_url
+
+
+class TeamsMessageSender:
+    def __init__(self, webhook_url):
+        self.webhook_url = webhook_url
+
+    def send_message(self, title, quote_text, quote_author, image_url):
+        "Method combines the whole message and send it to MSTeams channel"
+        myTeamsMessage = pymsteams.connectorcard(self.webhook_url)
+        myTeamsMessage.color(MESSAGE_COLOR)
+        myTeamsMessage.title(title)
+        myTeamsMessage.text(f"**Quote:** {quote_text}\n\n" 
+                            f"**Author:** {quote_author}\n\n"
+                            f"![Image]({image_url})")
+        try:
+            myTeamsMessage.send()
+            logging.info("Message successfully sent.")
+        except ConnectionError as e:
+            logging.error(f"Failed to send message due to connection error: {e}")
+        except HTTPError as e:
+            logging.error(f"Failed to send message due to HTTP error: {e}")
+        except TimeoutError as e:
```

Comment:
I would combine Timeout error, HttpError and ConnectionError to one case, describing troubles with a network